### ISSUE TITLE
Repository dispatch isn't working correctly

### ISSUE THREAD

**Issue by HardikSingh97** [peter-evans/create-pull-request#500]
### Issue Description
I am running a workflow in this branch of [hebi-hrdf](https://github.com/HebiRobotics/hebi-hrdf/tree/hardik/actions-setup). This workflow initiates a repository dispatch event that triggers a workflow in [this repo](https://github.com/HardikSingh97/hello-github-actions). The workflow that gets triggered here makes use of _create-pull-request_.

It seems to run successfully (here is a [log](https://github.com/HardikSingh97/hello-github-actions/runs/1001446804?check_suite_focus=true)) but no pull request actually gets created. 

In other words, for this same workflow:
- when triggered by a push, it runs successfully and creates a pull request.
- when triggered by a repository dispatch event, it runs successfully but does _not_ create a pull request.

I have tried a few different things but I can't seem to identify what's going on. Any guidance is greatly appreciated!

**Comment by peter-evans** [comment]
Hi @HardikSingh97 

I've only given this a brief look so far, but I think this might be a bug. I think these steps will reproduce it:

1. A pull request is created on branch `Automted-HRDF-Update`
1. The pull request is closed, **but the branch is not deleted**
1. The workflow runs again and finds that the `Automted-HRDF-Update` branch already exists. Furthermore, it doesn't need updating because it's already up to date with changes that were made in a previous run.
1. Because the branch didn't need updating the action thinks that all is fine and exits, but actually the pull request is closed.

So what should happen is that a new pull request should be created for the branch that still exists. I will work on a fix for this.

In the meantime, you can prevent this from happening by deleting the branch when you close the PR without merging it. I imagine this only happened because you were testing.

**Comment by peter-evans** [comment]
Fixed and released as `v3` and `v3.1.2`.
